### ISSUE TITLE
MR result of the models

### ISSUE THREAD

**Issue by msha096** [lw396285v/CSP.pytorch#1]
Hi,

Thanks for your sharing.

You uploaded the model with MR 11.76. Is that the result trained from the code with setting configure.teacher= False?

I ran the model you uploaded, the result is so different from yours.
Here is my result:
with net-26.pth, I got : Summerize: [Reasonable: 22.48%], [Bare: 14.10%], [Partial: 20.87%], [Heavy: 58.75%]
with net-89.pth, I got: Summerize: [Reasonable: 20.89%], [Bare: 13.67%], [Partial: 19.65%], [Heavy: 58.34%]

I have one 1080Ti gpu, batchsize is 2. 

Thanks!

**Comment by LewX** [comment]
I got same result
Summerize: [Reasonable: 20.89%], [Bare: 13.67%], [Partial: 19.65%], [Heavy: 58.34%]

**Comment by lwpyr** [comment]
I'm investigating in this now. I am going to check the model I uploaded.

**Comment by lwpyr** [comment]
Sorry, I forget to tell you something.
I found the real problem.
The resnet.py is not the one I am using, I will upload the repo now.
Here is my result of the best:
```
Dataset...
500/500
Net...
=== Frozen Parameter Layers ===
===============================
point:center
iter_per_epoch:2000
size_test:(1024, 2048)
down:4
alpha:0.999
radius:2
num_epochs:150
num_scale:1
use_horizontal_flips:True
add_epoch:0
scale:h
init_lr:0.0001
network:resnet50
val:True
img_channel_mean:[103.939, 116.779, 123.68]
test_path:./data/citypersons
teacher:False
size_train:(640, 1280)
offset:True
onegpu:4
train_random:False
caffemodel:False
brightness:(0.5, 2, 0.5)
gpu_ids:[0, 1]
val_frequency:1
train_path:./data/citypersons
Perform validation...
500/500 
Summerize: [Reasonable: 11.70%], [Bare: 7.81%], [Partial: 11.19%], [Heavy: 50.68%]
Validation time used: 147.016
```

**Comment by lwpyr** [comment]
And I will upload the entire repo when I finish my experiment.
I am conducting some new experiment right now, If the result is good I will upload these extra codes!

**Comment by msha096** [comment]
> Sorry, I forget to tell you something.
> I found the real problem.
> The resnet.py is not the one I am using, I will upload the repo now.
> Here is my result of the best:
> 
> ```
> Dataset...
> 500/500
> Net...
> === Frozen Parameter Layers ===
> ===============================
> point:center
> iter_per_epoch:2000
> size_test:(1024, 2048)
> down:4
> alpha:0.999
> radius:2
> num_epochs:150
> num_scale:1
> use_horizontal_flips:True
> add_epoch:0
> scale:h
> init_lr:0.0001
> network:resnet50
> val:True
> img_channel_mean:[103.939, 116.779, 123.68]
> test_path:./data/citypersons
> teacher:False
> size_train:(640, 1280)
> offset:True
> onegpu:4
> train_random:False
> caffemodel:False
> brightness:(0.5, 2, 0.5)
> gpu_ids:[0, 1]
> val_frequency:1
> train_path:./data/citypersons
> Perform validation...
> 500/500 
> Summerize: [Reasonable: 11.70%], [Bare: 7.81%], [Partial: 11.19%], [Heavy: 50.68%]
> Validation time used: 147.016
> ```

Hi, did you update all the layers' parameters to trainable? What is the different and the is there any rules of thumb for freezing layers? Thanks!

**Comment by lwpyr** [comment]
> > Sorry, I forget to tell you something.
> > I found the real problem.
> > The resnet.py is not the one I am using, I will upload the repo now.
> > Here is my result of the best:
> > ```
> > Dataset...
> > 500/500
> > Net...
> > === Frozen Parameter Layers ===
> > ===============================
> > point:center
> > iter_per_epoch:2000
> > size_test:(1024, 2048)
> > down:4
> > alpha:0.999
> > radius:2
> > num_epochs:150
> > num_scale:1
> > use_horizontal_flips:True
> > add_epoch:0
> > scale:h
> > init_lr:0.0001
> > network:resnet50
> > val:True
> > img_channel_mean:[103.939, 116.779, 123.68]
> > test_path:./data/citypersons
> > teacher:False
> > size_train:(640, 1280)
> > offset:True
> > onegpu:4
> > train_random:False
> > caffemodel:False
> > brightness:(0.5, 2, 0.5)
> > gpu_ids:[0, 1]
> > val_frequency:1
> > train_path:./data/citypersons
> > Perform validation...
> > 500/500 
> > Summerize: [Reasonable: 11.70%], [Bare: 7.81%], [Partial: 11.19%], [Heavy: 50.68%]
> > Validation time used: 147.016
> > ```
> 
> Hi, did you update all the layers' parameters to trainable? What is the different and the is there any rules of thumb for freezing layers? Thanks!

all parameters are trainable.
I use 2e-4 learning rate training for 100 epochs, then I just adjust the learning rate to 2e-5, and get this result.

If I freeze the first several layers, I can only about 13~14 MR.

anyway, you can try it on your machine. maybe you can find better results. you can even try SGD for extreme performance.

**Comment by msha096** [comment]
> init_lr:0.0001

Thank you! It shows in your terminal that init_learning rate is 1e-4. Which one is right?

**Comment by lwpyr** [comment]
> > init_lr:0.0001
> 
> Thank you! It shows in your terminal that init_learning rate is 1e-4. Which one is right?

I think 0.0001 for the first 50 epochs and 2e-5 for next 50 epochs is OK.
And I modify the network and get 11.07 MR with 0.16 sec per frame on 1080ti. (compare to original 11.7 MR with 0.3 sec per frame on 1080 ti, both implemented in Pytorch.)
I will update the repo several days later.

**Comment by msha096** [comment]
> > > init_lr:0.0001
> > 
> > 
> > Thank you! It shows in your terminal that init_learning rate is 1e-4. Which one is right?
> 
> I think 0.0001 for the first 50 epochs and 2e-5 for next 50 epochs is OK.
> And I modify the network and get 11.07 MR with 0.16 sec per frame on 1080ti. (compare to original 11.7 MR with 0.3 sec per frame on 1080 ti, both implemented in Pytorch.)
> I will update the repo several days later.

Hi,

I use your LR and I remove freeze layer, but the best MR is only 12.29%. Would you please tell me how you modify your network to get the better result? Thanks!

**Comment by lwpyr** [comment]
> > > > init_lr:0.0001
> > > 
> > > 
> > > Thank you! It shows in your terminal that init_learning rate is 1e-4. Which one is right?
> > 
> > 
> > I think 0.0001 for the first 50 epochs and 2e-5 for next 50 epochs is OK.
> > And I modify the network and get 11.07 MR with 0.16 sec per frame on 1080ti. (compare to original 11.7 MR with 0.3 sec per frame on 1080 ti, both implemented in Pytorch.)
> > I will update the repo several days later.
> 
> Hi,
> 
> I use your LR and I remove freeze layer, but the best MR is only 12.29%. Would you please tell me how you modify your network to get the better result? Thanks!

continue training or low the learning rate.
